server/utils/utils.ts

```diff
@@ -218,6 +218,17 @@ const isPassedArdOrCrd = (licence: LicenceSummary, prisoner: Prisoner | PrisonAp
   return false
 }
 
+const groupingBy = <T extends Record<K, unknown>, K extends keyof T>(arr: T[], keyField: K): T[][] => {
```

Comment:
Looks like some code wizardy here but can't really make sense of what is happening here?

Reply:
yes.. its a bit wizard-y we've just had a chat about it.